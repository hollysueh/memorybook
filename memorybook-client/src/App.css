/* Main App Styling   */
.App {
  padding-bottom: 200px;
  background-color: linen;
  color: rgb(107, 99, 90);
  font-family: 'Gill Sans', 'Gill Sans MT', Calibri, 'Trebuchet MS', sans-serif;
  font-size: 105%;
  text-align: center;
}
.pageHeading {
  margin-top: 30px;
  margin-bottom: 40px;
}

/* Header Page Styling */
.headerStyle {
  margin-top: 0px;
  margin-bottom: 0px;
  padding: 10px;
  background-color:rgb(141, 108, 108);
  font-size: calc(10px + 2vmin);
  color: white;
}
#headerNav {
  margin-top: 20px;
  margin-left: 5px;
  display: inline;
  float: left;
}
#headerImg {
  display: inline;
  float: center;
}
#headerGreet {
  margin-top: 35px;
  margin-right: 30px;
  display: inline-flex;
  float: right;
}
.headerLink {
  color: rgb(231, 203, 189);
  text-decoration-color: white;
  font-size: 90%;
}

/* NavBar Styling */
.navbar {
  background-color:rgb(141, 108, 108);
  border-bottom: 20px solid rgb(141, 108, 108);
  position: absolute;
}
#navDropdown {
  background-color:rgb(141, 108, 108);
  color: rgb(231, 203, 189);
  text-decoration-color: white;
  text-align: left;
}
.dropdownLink {
  color: rgb(107, 99, 90);
}
#styleDropdown {
  padding-bottom: 0px;
}

/* Landing (Home) Page Styling */
#landingHeader {
  margin: 40px;
  font-weight: 400;
  text-decoration: underline;
  font-size: 300%;
}
.landingSubHead {
  font-size: 140%;
}

/* Photo Album Styling ('my Memory Book') */
.pageNavBtn {
  margin: 30px;
  padding: 10px;
  border: 1px solid darkgray;
  border-radius: 5px;
  background-color:rgb(141, 108, 108);
  color: white;
  font-size: 120%;
}
#pagePrev {
  float: left;
}
#pageNext {
  float: right;
}
#pageAdd {
  float: center;
}
.flipBook {
 margin-left: 5%;
 margin-right: 5%;
}

/* Page Content Styling */
.pageContent {
  padding-top: 0px;
  padding-left: 20px;
  padding-right: 20px;
  padding-bottom: 40px;
  border: 1px solid darkgray;
  border-radius: 20px;
  box-shadow: inset 0 0 30px #000000;
}
.pageTxtInput {
  margin-top: 30px;
  margin-left: auto;
  margin-right: auto;
  padding: 10px;
  border: 1px solid darkgray;
  border-radius: 5px;
  background-color: linen;
  color:  rgb(59, 55, 50);
  text-align: center;
  display: block;
}
.pageImgInput {
  margin: 5px;
  color:  rgb(59, 55, 50);
  text-align: center;
}
.img {
  margin-left: auto;
  margin-right: auto;
  padding: 0px;
  display: block;
}
.insidePageBtn {
  margin: 5px;
  padding: 5px;
  border: 1px solid rgb(163, 152, 140);
  border-radius: 5px;
  background-color:darkgrey;
  color: white;
  font-size: 90%;
}
#savePg {
  padding: 10px;
  background-color:rgb(141, 108, 108);
  font-size: 90%;
}
#cancelPg {
  float: left;
}
#delPg {
  float: right;
}
#pgNo {
  float: center;
}
input[type='file'] {
  font-size: 70%;
}

/* FAQs Page Styling */
.FAQs {
  margin-left: 30px;
  text-align: left;
}
.question {
  font-weight: 600;
  text-decoration: underline;
}

/* Login / Create Account / Contact Page Styling */
.loginForm, .accountForm, .contactForm {
  text-align: left;
   column-width: 50%;
   margin-left: 30px;
   margin-right: 30px;
}
.formLabel {
  font-weight: 600;
}
.loginButton, .accountButton, .contactButton {
  padding: 10px;
  border: 1px solid darkgray;
  border-radius: 5px;
  background-color:rgb(141, 108, 108);
  color: white;
  font-size: 120%;
}

/* REFERENCES:
    input-file styling:
      https://stackoverflow.com/questions/36540105/how-to-style-a-choose-file-button-using-css-only
*/