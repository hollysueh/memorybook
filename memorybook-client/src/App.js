import React, { Component } from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import './App.css';

//Import components
import Header from './components/Header';
import Landing from './components/Landing';
import Login from './components/Login';
import CreateAccount from './components/CreateAccount';
import About from './components/About';
import PhotoAlbum from './components/PhotoAlbum';
import FAQs from './components/FAQs';
import Contact from './components/Contact';

//Render Components
class App extends Component {
  render() {
    return (
      <div className="App">
        <link
          rel="stylesheet"
          href="https://maxcdn.bootstrapcdn.com/bootstrap/4.2.1/css/bootstrap.min.css"
          integrity="sha384-GJzZqFGwb1QTTN6wy59ffF1BuGJpLSa9DkKMp0DgiMDm4iYMj70gZWKYbI706tWS"
          crossOrigin="anonymous"/>
        <BrowserRouter>
          <Header />
          <Switch>
          <Route path="/" exact={true} component={Landing} />
          <Route path="/Login" component={Login} />
          <Route path="/CreateAccount" component={CreateAccount} />
          <Route path="/About" component={About} />
          <Route path="/PhotoAlbum/:userID" component={PhotoAlbum} />
          <Route path="/FAQs" component={FAQs} />
          <Route path="/Contact" component={Contact} />
          </Switch>
        </BrowserRouter>
      </div>
    );
  }
}

export default App;