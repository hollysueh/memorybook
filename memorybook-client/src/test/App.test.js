import React from 'react';
import App from '../App';
import renderer from 'react-test-renderer';

//Test App.js renders correctly
test('renders correctly', () => {
  const tree = renderer
  .create(<App />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});
