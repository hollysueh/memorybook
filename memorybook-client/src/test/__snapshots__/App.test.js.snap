// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <link
    crossOrigin="anonymous"
    href="https://maxcdn.bootstrapcdn.com/bootstrap/4.2.1/css/bootstrap.min.css"
    integrity="sha384-GJzZqFGwb1QTTN6wy59ffF1BuGJpLSa9DkKMp0DgiMDm4iYMj70gZWKYbI706tWS"
    rel="stylesheet"
  />
  <div
    className="headerStyle"
  >
    <header>
      <div
        id="headerNav"
      >
        <div
          className="App"
        >
          <link
            crossOrigin="anonymous"
            href="https://maxcdn.bootstrapcdn.com/bootstrap/4.2.1/css/bootstrap.min.css"
            integrity="sha384-GJzZqFGwb1QTTN6wy59ffF1BuGJpLSa9DkKMp0DgiMDm4iYMj70gZWKYbI706tWS"
            rel="stylesheet"
          />
          <nav
            className="navbar navbar navbar-expand navbar-light"
          >
            <div
              className="dropdown nav-item"
              onKeyDown={[Function]}
            >
              <a
                aria-expanded={false}
                aria-haspopup={true}
                className="dropdown-toggle nav-link"
                href="#"
                id="navDropdown"
                onClick={[Function]}
                onKeyDown={[Function]}
                role="button"
              >
                Menu
              </a>
            </div>
          </nav>
        </div>
      </div>
      <div
        id="headerImg"
      >
        <img
          alt="logo"
          onClick={[Function]}
          src="memorybook_logo.png"
          width={250}
        />
      </div>
      <div
        id="headerGreet"
      >
        <h4>
          <div>
            <a
              className="headerLink"
              href="/Login"
              onClick={[Function]}
            >
              Login /
            </a>
            <a
              className="headerLink"
              href="/CreateAccount"
              onClick={[Function]}
            >
               Create Account
            </a>
          </div>
        </h4>
      </div>
    </header>
  </div>
  <div
    className="App"
  >
    <h1
      id="landingHeader"
    >
      Memory Book
    </h1>
    <p
      className="landingSubHead"
    >
      A 100% free virtual photo album maker.
    </p>
    <p
      className="landingSubHead"
    >
      <a
        href="/CreateAccount"
      >
        Click here
      </a>
       to get started.
    </p>
  </div>
</div>
`;
